ozaria/site/components/teacher-dashboard/BaseSingleClass/index.vue

```diff
@@ -69,14 +75,114 @@ export default {
       getActiveClassrooms: 'teacherDashboard/getActiveClassrooms',
       selectableStudentIds: 'baseSingleClass/selectableStudentIds',
       getSelectableOriginals: 'baseSingleClass/getSelectableOriginals',
-      classroomCourses: 'teacherDashboard/getCoursesCurrentClassroom'
+      classroomCourses: 'teacherDashboard/getCoursesCurrentClassroom',
+      aiScenarios: 'aiScenarios/getScenarios',
     }),
 
     modules () {
       // Reference below required to trigger a re-render when the refresh button is clicked.
       this.refreshKey // eslint-disable-line no-unused-expressions
 
       const selectedCourseId = this.selectedCourseId
+      const courseInstances = this.getCourseInstancesOfClass(this.classroom._id) || []
+
+      const assignmentMap = new Map()
+      for (const { courseID, members } of courseInstances) {
+        assignmentMap.set(courseID, new Set(members || []))
+      }
+
+      if (selectedCourseId === utils.courseIDs.HACKSTACK) {
+        const hackStackModuleNames = this.aiScenarios.reduce((acc, scenario) => {
+          acc.add(scenario.tool)
+          return acc
+        }, new Set())
+
+        const hackStackModels = [...hackStackModuleNames].map((moduleName, key) => {
+          const moduleNum = key + 1
+          const classSummaryProgress = []
+          const moduleScenarios = (this.aiScenarios || [])
+            .filter(scenario => scenario.tool === moduleName)
+
+          return {
+            moduleNum,
+            displayName: moduleName,
+            displayLogo: utils.aiToolToImage[moduleName] || null,
+            contentList: moduleScenarios
+              .map((scenario, index) => {
+                return {
+                  displayName: scenario.name,
+                  type: 'challengelvl',
+                  _id: scenario._id,
+                  tooltipName: `${index + 1}: ${scenario.name}`,
+                  description: '',
+                  contentKey: scenario._id,
+                  normalizedOriginal: scenario._id,
+                  normalizedType: 'challengelvl',
+                  contentLevelSlug: scenario.slug,
+                  isPractice: false
+                }
+              }),
+            studentSessions: this.students.reduce((studentSessions, student) => {
```

Comment:
can we instead use functions and make it more readable? I'm not a fan of functions with 50 lines